vendor/bat-native-ads/src/bat/ads/internal/ads/ad_events/search_result_ads/search_result_ad_event_handler.h

```diff
@@ -17,6 +19,11 @@ struct SearchResultAdInfo;
 
 namespace search_result_ads {
 
+using TriggerSearchResultAdEventCallback =
+    std::function<void(const bool,
+                       const std::string&,
+                       const mojom::SearchResultAdEventType event_type)>;
```

Comment:
This should be in `search_result_ad.h` and then we should add a new callback, maybe `SearchResultFireAdEventHandlerCallback`

Reply:
also due to name spacing we could have `search_result_ads::TriggerAdEventCallback`

Reply:
I've just renamed it to `search_result_ads::FireAdEventHandlerCallback` there is no callback needed in `search_result_ad.h`